test/test_ssl.rb

```diff
@@ -186,8 +186,103 @@ def test_add_certificate_multiple_certs
   end
 
   def test_add_certificate_chain_file
-    ctx = OpenSSL::SSL::SSLContext.new
-    assert ctx.add_certificate_chain_file(Fixtures.file_path("chain", "server.crt"))
+    # Create chain certificates file
+    GC.disable # for tempfile
```

Comment:
I don't believe this is needed, `certs` or `pkey` shouldn't be GC'ed before the `#close` calls in the `ensure` block?

https://ruby-doc.org/stdlib-2.5.0/libdoc/tempfile/rdoc/Tempfile.html also recommends calling `#unlink`.

Reply:
Thank you!! Fixed.